test/lifecycle_test/component2.dart

```diff
@@ -29,8 +29,9 @@ class _SetStateTest extends react.Component2 with LifecycleTestHelper {
       };
 
   @override
-  componentWillReceiveProps(_) {
-    recordLifecyleCall('componentWillReceiveProps');
+  Map getDerivedStateFromProps(_, __) {
+    recordLifecyleCall('getDerivedStateFromProps');
+    return new Map();
```

Comment:
#nit prefer collection literals
```suggestion
return {};
```